Tests/XCUITests/HistoryTests.swift

```diff
@@ -103,215 +98,214 @@ class HistoryTests: BaseTestCase {
-        navigator.goto(TabTray)
-        navigator.performAction(Action.AcceptRemovingAllTabs)
+        openBookOfMozilla()
 
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
+        closeFirstTabByX()
+
+        // Long tap a recently closed item launches a context menu
+        navigator.nowAt(HomePanelsScreen)
         navigator.goto(HistoryRecentlyClosed)
         waitForExistence(app.tables["Recently Closed Tabs List"])
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
-        app.tables.cells.staticTexts[closedWebPageLabel].press(forDuration: 1)
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+        app.tables.cells.staticTexts[bookOfMozilla["label"]!].press(forDuration: 1)
         waitForExistence(app.tables["Context Menu"])
         XCTAssertTrue(app.tables.otherElements[ImageIdentifiers.newTab].exists)
         XCTAssertTrue(app.tables.otherElements[ImageIdentifiers.newPrivateTab].exists)
     }
 
     func testOpenInNewTabRecentlyClosedItem() {
-        // Open the default website
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        waitForTabsButton()
+        openBookOfMozilla()
+
+        closeFirstTabByX()
+
+        // Ensure the tab is closed
         navigator.goto(TabTray)
-        navigator.performAction(Action.AcceptRemovingAllTabs)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(HistoryRecentlyClosed)
-        waitForExistence(app.tables["Recently Closed Tabs List"])
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
         let numTabsOpen = userState.numTabs
         XCTAssertEqual(numTabsOpen, 1)
-        app.tables.cells.staticTexts[closedWebPageLabel].press(forDuration: 1)
+        XCTAssertFalse(app.staticTexts[bookOfMozilla["title"]!].exists)
+
+        // Open the page on a new tab from History Recently Closed screen
+        navigator.nowAt(HomePanelsScreen)
+        navigator.goto(HistoryRecentlyClosed) // Note: A new tab is open before opening the history page, so there'll be 3 tabs opened at the end.
+        waitForExistence(app.tables["Recently Closed Tabs List"])
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+        app.tables.cells.staticTexts[bookOfMozilla["label"]!].press(forDuration: 1)
         waitForExistence(app.tables["Context Menu"])
         app.tables.otherElements[ImageIdentifiers.newTab].tap()
+        app.buttons["Done"].tap()
+
+        // The page is opened on the new tab
+        navigator.nowAt(HomePanelsScreen)
         navigator.goto(TabTray)
-        let numTabsOpen2 = userState.numTabs
-        XCTAssertEqual(numTabsOpen2, 2)
+        XCTAssertTrue(app.staticTexts[bookOfMozilla["title"]!].exists)
+        XCTAssertEqual(userState.numTabs, 3)
```

Comment:
Note that line 200 opens a new tab before navigating to the "Recently Closed" screen no matter what has taken place prior. That's why by here we've 3 tabs open. Perhaps the `FxScreenGraph` needs some work but I have not identified where yet.

Also, I've discovered that `userState.numTabs` is accurate only when the `TabTray` is opened currently. We should not query this value when we are not on the `TabTray`.